int32[] ee_ids
int32[] contact_surface_ids

int32 knot_nums
float64 knot_time